lms/djangoapps/certificates/views/webview.py

```diff
@@ -588,46 +546,131 @@ def render_html_view(request, user_id, course_id):
             course_id,
             user_id,
         )
-        return render_to_response(invalid_template_path, context)
+        return _render_invalid_certificate(course_id, platform_name, configuration)
+
+    # Get data from Discovery service that will be necessary for rendering this Certificate.
+    catalog_data = _get_catalog_data_for_course(course_key)
+
+    # If custom templates are enabled, see if there's one associated with this course/org.
+    custom_template = None
+    if settings.FEATURES.get('CUSTOM_CERTIFICATE_TEMPLATES_ENABLED', False):
+        custom_template = get_certificate_template(course.id, user_certificate.mode, catalog_data.get('content_language'))
 
-    context['certificate_data'] = active_configuration
+    user_language = translation.get_language()
+    certificate_language = user_language
 
-    # Append/Override the existing view context values with any mode-specific ConfigurationModel values
-    context.update(configuration.get(user_certificate.mode, {}))
+    # If we're using a custom template that is associated with a particular language, we'll need to make sure to
+    # translate the context variables that will be used to render that template in that language.
+    if custom_template and custom_template.language:
+        certificate_language = _get_closest_released_language(custom_template.language)
 
-    # Append organization info
-    _update_organization_context(context, course)
+    with translation.override(certificate_language):
```

Comment:
Just want to ensure that if we pass in garbage here for some reason that nothing breaks since `_get_closest_released_language` will return back if there's no match.  When we override with a non-existent language will it just default to English anywhere we call `gettext` since there is no file for a garbage match?